src/effects/lv2/LV2Validator.cpp

```diff
@@ -705,12 +705,24 @@ void LV2Validator::SetSlider(
    ctrl.slider->SetValue(lrintf((val - lo) / (hi - lo) * 1000.0));
 }
 
+void LV2Validator::UpdateControlPortValue(
+   LV2EffectSettings& settings, size_t controlPortIndex, float value)
+{
+   const auto currentValue = settings.values[controlPortIndex];
+
+   if (std::abs(currentValue - value) < std::numeric_limits<float>::epsilon())
```

Comment:
Similar concerns here.  Some controls are discrete -- toggles or choices.  Maybe the right epsilon for the control should be an argument of the function.